PayloadPkg/OsLoader/OsLoader.c

```diff
@@ -670,6 +676,27 @@ BeforeOSJump (
   }
   AddMeasurePoint (0x4100);
 
+  // Add the measurepoint timestamp to the HOB so that it can be updated in the ACPI table
+  PerfData = GetPerfDataPtr();
+  GuidHob = GetNextGuidHob (&gLoaderPerformanceInfoGuid, (VOID *)(UINTN)PcdGet32 (PcdPayloadHobList));
+  if (GuidHob != NULL) {
+    PerfInfo            = (PERFORMANCE_INFO *)GET_GUID_HOB_DATA (GuidHob);
+    PerfInfo->TimeStamp[PerfInfo->Count] = PerfData->TimeStamp[PerfData->PerfIndex-1];
+  }
+
+  // Add perf data to FPDT - SBL Performance Table
+  SystemTableInfo = GetSystemTableInfo ();
+  if ((SystemTableInfo != NULL) && (SystemTableInfo->AcpiTableBase != 0)) {
+    Rsdp = (EFI_ACPI_3_0_ROOT_SYSTEM_DESCRIPTION_POINTER *)(UINTN)SystemTableInfo->AcpiTableBase;
+    Fpdt = FindAcpiTableBySignature((EFI_ACPI_DESCRIPTION_HEADER *)(UINTN)Rsdp->RsdtAddress, SIGNATURE_32('F', 'P', 'D', 'T'), NULL);
+    if (Fpdt != NULL) {
+      DEBUG ((DEBUG_INFO, "FPDT found @ 0x%X, updating..\n", Fpdt));
+      PlatformService->AcpiTableUpdate((UINT8 *)Fpdt, Fpdt->Length);
```

Comment:
I remember it would append a new ACPI table with this call. If so, will it have 2 FPDT tables?
Could we just update the existing FPDT table during the EndOfFirmware notification? this way we don't need any change in OsLoader.

Reply:
The existing FPDT table is updated in this call. There will only be 1 FPDT.

Reply:
ok. the existing FPDT is updated instead of appending a new table.
For the second question, it could make the Osloader payload clean if we could do it in stage2 in the callback of EndOfFirmware.